File: cxx/testing/print_var_names_test.cxx
#include <bmi.hxx>

#include <iostream>
#include <stdio.h>
#include <stdlib.h>

void print_var_names (BMI::Model model);

int
main (void)
{
  int i;
  const int n_steps = 10;
  BMI::Model model;

  model.initialize ("");

  std::cout << model.get_component_name () << std::endl;

  print_var_names (model);

  model.finalize ();

  return EXIT_SUCCESS;
}

void
print_var_names (BMI::Model model)
{
  char **var_names = NULL;
  char **name;

  var_names = (char**)model.get_input_var_names ();
  fprintf (stdout, "Input var names\n");
  fprintf (stdout, "===============\n");
  for (name = var_names; *name; name++)
    fprintf (stdout, "%s\n", *name);
  fprintf (stdout, "\n");

  var_names = (char**)model.get_output_var_names ();
  fprintf (stdout, "Output var names\n");
  fprintf (stdout, "================\n");
  for (name = var_names; *name; name++)
    fprintf (stdout, "%s\n", *name);
  fprintf (stdout, "\n");

  return;
}

File: bindings/python/testing/print_var_names_test.py
#!/usr/bin/env python
from __future__ import print_function

import sys
import numpy as np

from poisson import BmiPoisson


def print_var_names(model):
    print('Input var names: ', end='')
    names = model.get_input_var_names()
    names.sort()
    print('%s' % ', '.join(names))

    print('Output var names: ', end='')
    names = model.get_output_var_names()
    names.sort()
    print('%s' % ', '.join(names))


def main():
    model = BmiPoisson()

    model.initialize()

    print('%s' % model.get_component_name())

    print_var_names(model)

    model.finalize()


if __name__ == '__main__':
    main()
